### DISCUSSION TITLE
Is there an elegant way to launch wezterm from file explorer in windows?

### DISCUSSION THREAD

**Discussion by abceleung** [wezterm/wezterm#5015] [Q&A]
### What Operating System(s) are you running on?

Windows

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

Hi, I used Windows Terminal before Wezterm. In file explorer, I can spawn a new Windows Terminal instance by typing "wt" on the address bar of file explorer and press enter.

I want to replicate this behavior with Wezterm. Currently I use a `wt.cmd` file which contains
```
start wezterm-gui -e --cwd .
```
and put the file on path. Then by typing `wt` on file explorer's address bar and pressing enter, a cmd window is spawned (and closed immediately), after that a Wezterm instance is spawned.

Is there a way to skip the cmd window and spawn Wezterm directly? Thanks!

([This](https://wezfurlong.org/wezterm/cli/general.html) part of docs hints it may be possible, but I don't know how)

**Comment by MrDwarf7** [comment]
Best way to do this honestly - 
If you have it installed via scoop, go to the `~\scoop\shims\` dir, and take the `wezterm-gui.exe` & `wezterm-gui.shim` and make a copy of them.

Then just rename them to whatever you want to use. I did this and called mine wzt so I can still open wt (If I ever need or want to again in the future that is)

**Comment by TheAjaykrishnanR** [comment]
I wrote a small dotnet console application just for this:

```csharp
using System.Diagnostics;

public class WeztermStartup
{
	public static void Main()
	{
		string cwd = Directory.GetCurrentDirectory();
		ProcessStartInfo psi = new();
		psi.CreateNoWindow = true;
		psi.UseShellExecute = false;
		psi.RedirectStandardOutput = true;
		psi.WindowStyle = ProcessWindowStyle.Hidden;
		psi.FileName = "wezterm.exe";
		psi.Arguments = $"start --cwd {cwd}";
		Process process = new();
		process.StartInfo = psi;
		process.Start();
	}
}
```

compile it using:
```
 dotnet publish --self-contained false -p:PublishSingleFile=true
```

rename the file as you wish and keep it alongside the `wezterm.exe` executable and you can launch it from the address bar (wezterm directory must be in PATH)

**Comment by MrDwarf7** [comment]
@abceleung If you're wanting an addition to the right-click menu I had a look into how to do it. 
It's fairly straightforward honestly, made a little Windows reg key to do it: 

```reg

Windows Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\Directory\Background\shell\Wezterm]
@="&Wezterm"
"Icon"="C:\\Users\\<<USERNAME>>\\scoop\\apps\\wezterm\\current\\wezterm-gui.exe"

[HKEY_CLASSES_ROOT\Directory\Background\shell\Wezterm\command]
@="C:\\Users\\<<USERNAME>>\\scoop\\apps\\wezterm\\current\\wezterm-gui.exe start --cwd %v."

```

Replace the '<\<USERNAME>\>' with your username, no quotes & removing the double angled brackets on either side also.

If your path is different (i.e: GitHub source download and built locally for instance), then simply replace the paths with the _full_ path to the wezterm-gui.exe executable.

There's also [this](https://github.com/ScoopInstaller/Extras/blob/354a5a3208668934854bc0f9879a79d6d1bceda2/bucket/wezterm.json#L17C4-L27C7) in the scoop installer. 
It appears to use local paths for your specific user, which may be more desirable and is more in-line with most programs these days.

When I checked my own manifest.json it didn't even have the post_install in it, not sure why.
